presto-main/src/main/java/com/facebook/presto/cost/StatisticRange.java

```diff
@@ -97,7 +97,7 @@ public double overlapPercentWith(StatisticRange other)
         double lengthOfIntersect = min(high, other.high) - max(low, other.low);
         if (isInfinite(lengthOfIntersect)) {
             if (isFinite(this.distinctValues) && isFinite(other.distinctValues)) {
-                return min(this.distinctValues, other.distinctValues) / max(this.distinctValues, other.distinctValues);
+                return min(other.distinctValues / this.distinctValues, 1);
```

Comment:
When one range has {{n}} NDV and another {{m > n}}, we assume (elsewhere) that {{n}} values match. So {{theSmallerRange.percentOverlapWith(theBiggerRange)}} should return {{1.}}. Previously it would return {{n/m < 1}}.